src/security/srv_acl.c

```diff
@@ -506,16 +506,65 @@ get_sec_capas_for_token(Auth__Token *token, struct ownership *ownership,
 	return rc;
 }
 
+static int
+get_sec_origin_for_token(Auth__Token *token, char **machine)
+{
+	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
+	int 			machine_size = 0;
+	int 			rc = 0;
+	char 			*mtmp;
+	Auth__Sys		*authsys;
+
+
+	if (token == NULL || machine == NULL) {
+		D_ERROR("NULL input\n");
+		return -DER_INVAL;
+	}
+
+	rc = get_auth_sys_payload(token, &authsys);
+	if (rc != 0)
+		return rc;
+
+	if(authsys->machinename) {
+		 machine_size = strnlen(authsys->machinename, MAXHOSTNAMELEN);
+	} else {
+		D_ERROR("Malformed AuthSys token missing machinename");
+		rc = -DER_INVAL;
+		goto out;
+	}
+
+	if (machine_size > 64) {
```

Comment:
I really think these magic numbers should go away entirely. If we need to check length against MAXHOSTNAMELEN let's use the constant.

Reply:
+1

Reply:
That makes sense. Fixed.